preview/MsixCore/msixmgr/Package.cpp

```diff
@@ -148,30 +155,120 @@ HRESULT MsixCoreLib::PackageBase::ProcessPSFIfNecessary()
+            {
+                std::string id = (*itr)["id"].GetString();
+                std::wstring jsonApplicationId;
+                jsonApplicationId.assign(id.begin(), id.end());
+
+                //ApplicationId in json should match applicationId from the xml manifest. Right now, we only process first application from the xml manifest anyway
+                if (CaseInsensitiveEquals(jsonApplicationId, m_applicationId))
+                {
+                    // Fail if endScript is present in current implementation as we do not support those cases yet
+                    if ((*itr).HasMember("endScript"))
+                    {
+                        return HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED);
+                    }
+
+                    std::string executable = (*itr)["executable"].GetString();
+                    std::wstring psfExecutable;
+                    psfExecutable.assign(executable.begin(), executable.end());
+
+                    std::string arguments = (*itr)["arguments"].GetString();
+                    std::wstring psfArguments;
+                    psfArguments.assign(arguments.begin(), arguments.end());
+
+                    std::string workingDirectory = (*itr)["workingDirectory"].GetString();
+                    std::wstring psfWorkingDirectory;
+                    psfWorkingDirectory.assign(workingDirectory.begin(), workingDirectory.end());
+
+                    m_executionInfo.commandLineArguments = psfArguments;
+
+                    // resolve the given paths from json into full paths.
+                    m_executionInfo.resolvedExecutableFilePath = FilePathMappings::GetInstance().GetExecutablePath(psfExecutable, m_packageFullName.c_str());
+                    m_executionInfo.workingDirectory = FilePathMappings::GetInstance().GetExecutablePath(psfWorkingDirectory, m_packageFullName.c_str());
+
+                    if ((*itr).HasMember("startScript"))
+                    {
+                        const Value& startScript = (*itr)["startScript"];
+
+                        std::string scriptPath = startScript["scriptPath"].GetString();
+                        std::wstring psfScriptPath;
+                        psfScriptPath.assign(scriptPath.begin(), scriptPath.end());
+                        m_scriptSettings.scriptPath = psfScriptPath;
+
+                        if (startScript.HasMember("runOnce"))
+                        {
+                            bool runOnce = startScript["runOnce"].GetBool();
+                            m_scriptSettings.runOnce = runOnce;
+
+                            if (runOnce == false)
+                            {
+                                return HRESULT_FROM_WIN32(ERROR_NOT_SUPPORTED);
+                            }
+                        }
+                        else
+                        {
+                            m_scriptSettings.runOnce = false;

```

Comment:
>false [](start = 55, length = 5)

this one defaults to true.

Reply:
Removed else case since its initialized now. If name-value pair is absent, it will default to respective value.